server/game/core/card/Card.ts

```diff
@@ -310,6 +310,10 @@ export class Card extends OngoingEffectSource {
         return this.type === CardType.TokenUnit || this.type === CardType.TokenUpgrade;
     }
 
+    public isTokenUnit(): this is TokenCard {
```

Comment:
Should be `this is TokenUnitCard`. If we're adding this we should also add `isTokenUpgrade()` for completeness

Reply:
yep, good catch. Added the additional method.